docs: Protecting TypeScript NestJS Web API Template

djnovin opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

In this pull request, I've integrated Unkey to provide route protection and rate limiting for the application for hackathon. This is a side quest to make a template using a framework.

API Key Verification: Added middleware to verify API keys using Unkey. Only requests with valid API keys in the Authorization header (Bearer token) are allowed to access protected routes.

Fixes #2361 

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

<!-- If there isn't an issue for this PR, please re-review our Contributing Guide and create an issue -->

## Type of change

<!-- Please mark the relevant points by using [x] -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

```bash
git clone https://github.com/djnovin/unkey-ts-nestjs
```

```bash
pnpm install
```

```bash
pnpm run start
```

```bash
curl http://localhost:3000/api/v1/protected -H "Authorization: Bearer <YOUR_API_KEY>"
```

## Checklist

<!-- We're starting to get more and more contributions. Please help us making this efficient for all of us and go through this checklist. Please tick off what you did  -->

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Added a new project entry demonstrating the integration of Unkey with TypeScript and NestJS.
	- Included a link to the corresponding GitHub repository for user reference.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

chronark commented:
```ts
private async validateBearerToken(token: string): Promise<boolean> {
    const validationResponse = await verifyKey(token);

    if (!validationResponse) return false;

    if (validationResponse.error) return false;

    return true;
  }
 ```

This is completely insecure and does not check the status of verification at all.
You must check the `validationResponse.valid` property

Also looks like you copy pasted from another template, so your setup and testing instructions are wrong

djnovin commented:
> ```ts
> private async validateBearerToken(token: string): Promise<boolean> {
>     const validationResponse = await verifyKey(token);
> 
>     if (!validationResponse) return false;
> 
>     if (validationResponse.error) return false;
> 
>     return true;
>   }
> ```
> 
> This is completely insecure and does not check the status of verification at all. You must check the `validationResponse.valid` property
> 
> Also looks like you copy pasted from another template, so your setup and testing instructions are wrong

Updated to utilise `result.valid`.

```ts

  private async validateBearerToken(token: string): Promise<boolean> {
    try {
      const { result } = await verifyKey(token);

      return result.valid;
    } catch (err) {
      return false;
    }
  }
  
  ```
  
  I have also updated PR instructions to point to correct `git` repo.
